### PR TITLE
Fix long repr for Categorical

### PR DESCRIPTION
**Pull request by jcrist**

### CODE REVIEW

**Comment by cpcloud** [comment]
awesome thx ... looking now

File: datashape/coretypes.py [hunk 1]
```diff
@@ -1038,8 +1037,9 @@ def __init__(self, categories, type=None):
         self.type = type or datashape.discover(self.categories).measure
```
Comment (cpcloud) [inline datashape/coretypes.py]: what are your thoughts on `type` here? do you think it's necessary?

Comment (cpcloud) [inline datashape/coretypes.py]: IMO i don't think we should add it now ... and only add it if someone provides a use case for it

Comment (cpcloud) [inline datashape/coretypes.py]: i'm happy to do the work to remove it BTW ... just wondering what you think

Comment (jcrist) [inline datashape/coretypes.py]: I'm fine with it either way. I don't need this functionality, but someone might.

**Pull request merged**